### PR TITLE
Resolve dataset preparation discrepancy for elasticity, darcy problems

### PR DESCRIPTION
**Pull request by vpuri3**
Hi LNO Authors,

I understand that you have based your experimental setup on that of [Transolver](https://github.com/thuml/Transolver), and that you make direct comparison with their results in your [paper](https://arxiv.org/abs/2406.03923). As such, I wanted to inform you of discrepancies in the data preparation stage between your experimental setup and that of Transolver which result in substantial changes in validation loss.

## Elasticity dataset

|            | Data File                    | Num Train | Num Test |
|------------|------------------------------|-----------|----------|
| Transolver | Random_UnitCell_sigma_10.npy | 1000      | 200      |
| LNO.       | Random_UnitCell_sigma_10.npy | 1000      | 1000     |

Transolver (based on [Transolver/PDE-SolvingStandardBenchmark/exp_elas.py](https://github.com/thuml/Transolver/blob/main/PDE-Solving-StandardBenchmark/exp_elas.py)) uses 200 cases for the test set, whereas LNO uses 1000.

On master:
```
$ ./scripts/LNO_Elasticity.sh
Epoch: 500      Learning Rate :1.0246740125861636e-11   Train Loss: 0.0040976484306156635   Val Loss: 0.005123566836118698
Finish Training !
```

On this branch:
```
$ ./scripts/LNO_Elasticity.sh
Epoch: 500      Learning Rate :1.0246740125861636e-11   Train Loss: 0.0040976484306156635   Val Loss: 0.01531931385397911
Finish Training !
```

This dramatically changes the validation loss from `0.0051` to `0.0153`.

## Darcy dataset

|            | Data File                             | Training Resolution| Num Train | Num Test |
|------------|---------------------------------------|--------------------|-----------|----------|
| Transolver | piececonst_r421_N1024_smooth{1,2}.mat | 85 x 85            | 1000      | 200      |
| LNO.       | piececonst_r241_N1024_smooth{1,2}.mat | 241 x 241          | 1024      | 1024     |

Transolver (based on [Transolver/PDE-SolvingStandardBenchmark/exp_darcy.py](https://github.com/thuml/Transolver/blob/main/PDE-Solving-StandardBenchmark/exp_darcy.py)) and LNO use different dataset files, different train/test splits, and training resolutions.

On master:
```
$ ./scripts/LNO_Darcy.sh
Epoch: 500      Learning Rate :1.3764955254324295e-11   Train Loss: 0.003548223525285721        Val Loss: 0.005500853061676025
Finish Training !
```

On this branch:
```
$ ./scripts/LNO_Darcy.sh
Epoch: 500      Learning Rate :1.401177331784148e-11    Train Loss: 0.008193107321858406        Val Loss: 0.009375382214784622
Finish Training !
```

This changes the validation loss form `0.0055` to `0.0093`.

---
We would greatly appreciate it if you could share the code for all the comparison models that were used. Running them within a unified codebase would help ensure consistency and avoid similar discrepancies. Additionally, the preparation scripts for other datasets may also contain similar issues that would benefit from a thorough examination.

Best,
Vedant Puri

### CODE REVIEW

**Comment by L-I-M-I-T** [comment]
Thank you for your interest in our work! 

For the Elasticity problem, since ```Random_UnitCell_sigma_10.npy``` contains 2000 samples, we selected 1000 samples for training (the same setting as Transolver) and used the remaining 1000 samples for testing. We believe that using more test samples can more accurately reflect the results from a statistical perspective.

For the Darcy problem, our experiments mainly used a resolution of 241×241 with a 1024/1024 train/test split. Under this setting, we evaluated the performance of baselines such as Transolver and GNOT, and found that the results were consistent with those reported in their papers. Therefore, we could directly cite the experimental results of other methods from their reported tables.

**Comment by vpuri3** [comment]
Thank you for the clarification.

Just to reiterate where my concern lies — **in order for comparisons to be fair and meaningful, all models need to be evaluated using the exact same train/test split and dataset configuration**. If, for example, LNO is being evaluated on a larger or different test set than Transolver (or vice versa), the reported validation losses are not directly comparable — regardless of model quality.

In the Elasticity case, re-running your script with the Transolver split worsens the validation loss by `2.9x`, highlighting that the split itself has a major impact. The same issue appears in the Darcy setup, where both the dataset resolution and split differ.

For a fair comparison — especially since your paper directly reports improvements over Transolver in Table 1 — all models (including the baselines) should be run under identical data splits and preprocessing steps. This principle applies broadly in benchmarking and reproducibility.

Could we align on a shared set of splits for all test cases and re-run the baselines and LNO under those conditions? I'd be happy to help or contribute scripts if needed.

**Comment by L-I-M-I-T** [comment]
Thank you for your reiteration, I completely understand your concern. In the early stage of this work, we had already compared Transolver with our method on the Elasticity and Darcy problem using the same train/test split and the same resolution, so the performance comparison here has already been aligned. Table 5 in [Transolver++](https://arxiv.org/pdf/2502.02414) provides a more intuitive comparison after alignment.

**Comment by vpuri3** [comment]
Thank you for your clarification and for pointing toward Transolver++ Table 5.

That said, I want to clearly point out that the default settings in the public LNO repository still use a different train/test split and, in the case of Darcy, a different resolution, compared to the evaluation setup used by Transolver. As shown above, simply switching to the Transolver split on the same code changes the validation loss from 0.0051 to 0.0153 on Elasticity — a ~3× difference. This highlights that the dataset split itself is driving much of the improvement in the reported metrics.

While I understand that aligned comparisons may have been run internally during the development phase, the results shown in your NeurIPS paper (Table 1) appear to be using the misaligned settings by default, and the released code reproduces those numbers. This makes it difficult for others to reproduce a fair comparison or to trust that LNO is strictly outperforming Transolver when evaluated under identical conditions.

For fairness and reproducibility — especially when making direct claims of improvement over other methods — all models should be trained and evaluated on the same split, same resolution, and same preprocessing. I hope the LNO team can either:

- provide scripts that reproduce the aligned experiments using the same splits as Transolver, or

- explicitly clarify in the repo and paper which results correspond to aligned vs. misaligned splits.

This would go a long way toward ensuring transparency and helping the community correctly interpret LNO’s improvements.

I appreciate your time, and I’m happy to help test or sync configurations if needed.

**Comment by L-I-M-I-T** [comment]
Thanks for pointing out the difference between our experimental setting and Transolver offical codes (which released after the submission of our paper). 

In conclusion, we emphasize that in our experiments, **all settings of Transolver and LNO are exactly the same**. Specifically, for Elasticity, we use 1000 samples for training and another 1000 samples for testing, while for Darcy, we use a resolution of 241 × 241.

Therefore, the comparison between LNO and Transolver in our paper is fair and reproducible.

At the time of submitting this paper (May 22, 2024), the official Transolver codes were not available to the public. We had to set up the experiments with our best effort to fit the description in the paper or follow other work (e.g., GNOT).

Finally, we believe that a larger test set (1000 rather than 200) and a higher resolution (241 rather than 85) are better for revealing the real model performance. We encourage the community to test existing models under more challenging settings.

**Comment by vpuri3** [comment]
Thank you for the clarification and discussion. I appreciate you taking the time out to elaborate on this matter for me. We’ll make sure to re-benchmark both methods under consistent splits in our own work, and will document this discrepancy for transparency. Best regards,
